### PR TITLE
Specialist CSC108 React Component

### PR DESCRIPTION
**Pull request by christinem**
This is a first dive into transitioning Check My Post to React. At the moment, this inserts the CSC108 Specialist React component into the `spec_csc108` div, and it updates whether it is "active" or not on page launch or refresh (as it did before).

### CODE REVIEW

File: hs/Scripts.hs [hunk 1]
```diff
@@ -89,8 +90,14 @@ postScripts = sequence_ (map toScript [
                                           "/static/js/graph/objects/node.js",
                                           "/static/js/post/fill_textboxes.js",
                                           "/static/js/graph/create_data.js",
-                                          "/static/js/post/update_categories.js"])
-
+                                          "/static/js/post/update_categories.js",
+                                          "static/js/common/course_description.js",
```
Comment (Ian-Stewart-Binks) [inline hs/Scripts.hs]: Missing the leading `/`.

Comment (Ian-Stewart-Binks) [inline hs/Scripts.hs]: Whoops, this looks like an internal PR. Nevermind me.

Comment (david-yz-liu) [inline hs/Scripts.hs]: @Ian-Stewart-Binks That's fine, your comment is still valid. @christinem please add the `/`.

File: public/js/post/post.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+var CourseCode = React.createClass({
+    getInitialState: function() {
+        return {selected: false}
+    },
+
+    componentWillMount: function() {
+        var isSelected = (getCookie(this.props.courseID) == 'active') ? true : false;
```
Comment (david-yz-liu) [inline public/js/post/post.js]: I imagine the `? true : false` is redundant; just return the value of the comparison. Also, use `===` not `==`.

File: public/js/post/post.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        }
+
+        return (
+            <div id = {this.props.courseID} className = "course">
+                <p className = {categoryClasses} onClick={this.toggleFullInfo}>  {this.props.courseID.toUpperCase() + "H"}  </p>
+                <div id = {this.props.courseID + '_info'} className = 'more-info'>
+                    <p className = {courseClasses}> {getCourseTitle(this.props.courseID)} </p>
+                </div>
+            </div>
+
```
Comment (david-yz-liu) [inline public/js/post/post.js]: Remove this blank line.

File: public/js/post/post.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        var categoryClasses = 'code';
+        var courseClasses = 'full_name';
+
+        if (this.state.selected) {
+            categoryClasses += ' category_selected';
+            courseClasses += ' course_selected';
+        }
+
+        return (
+            <div id = {this.props.courseID} className = "course">
```
Comment (david-yz-liu) [inline public/js/post/post.js]: I think we're not putting spaces around `=` inside HTML.

File: public/js/post/post.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+                    <p className = {courseClasses}> {getCourseTitle(this.props.courseID)} </p>
+                </div>
+            </div>
+           
+        );
+    }
+})
+
+
+React.render(<CourseCode courseID = 'csc108'/>, document.getElementById('spec_csc108'))
```
Comment (david-yz-liu) [inline public/js/post/post.js]: Make sure you end the file with a new line.

File: hs/Scripts.hs [hunk 6]
```diff
@@ -89,8 +90,14 @@ postScripts = sequence_ (map toScript [
...
+                                          "/static/js/common/objects/course.js",
+                                          "/static/js/common/objects/section.js",
+                                          "/static/js/common/utilities/util.js",
+                                          "https://cdnjs.cloudflare.com/ajax/libs/react/0.13.1/react.js",
+         "https://cdnjs.cloudflare.com/ajax/libs/react/0.13.1/JSXTransformer.js"])
```
Comment (david-yz-liu) [inline hs/Scripts.hs]: I've just added some global javascript includes. Please remove the two React scripts here; they should be automatically loaded.

Note: you'll need to do a pull from `upstream/react-post` to get the updates.

**Comment by david-yz-liu** [comment]
Nice work, @christinem. I really do like the new green.

**Comment by christinem** [comment]
@david-yz-liu made all the requested changes. And thanks, glad you liked it!

**Comment by david-yz-liu** [comment]
Thanks, @christinem!

**Pull request merged**